assets/app/scripts/services/data.js

```diff
@@ -488,17 +487,11 @@ angular.module('openshiftConsole')
 
     // For now just watch the type, eventually we may want to do something more complicated
     // and watch just the object if the type is not already being watched
-    this.watch(resource, context, null, opts);
-
-    // returned handle needs resource, name, context, and callback in order to unwatch
-    return {
-      resource: resource,
-      name: name,
-      context: context,
-      callback: wrapperCallback,
-      objectCallback: callback,
-      opts: opts
-    };
+    var handle = this.watch(resource, context, wrapperCallback, opts);
+    handle.objectCallback = callback;
+    handle.name = name;
+    
+    return handle;
   };
 
   DataService.prototype.unwatch = function(handle) {
```

Comment:
name wouldnt normally be on the handle, thats what name is here, would you rather i rename it to objectName?